### ISSUE TITLE
Registering multiple persons fails in TYPO3 8.7

### ISSUE THREAD

**Issue by k-pd** [oliverklee/ext-seminars#131]
I setup a simple testing site to make sure all features required for our update are working as expected. I followed the documentation and succeeded with a single person registration (Seats to book: 1, Persons to register: X myself). Trying the same registration process with more seats and additional names (with or without `X myself`) fails. After clicking "Proceed to checkout" all names are shown correctly, but the console is showing a JSON.parse error (see screenshot). Clicking the submit button again sends the form values (including unusable JSON data in tx_seminars_pi1_registration_editor[structured_attendees_names], see screenshot), redirecting me always back to the registration form, stating: "Please provide the names of all persons to register."

I'm sure, this has to do with Typo3 V8.7 always HTML encoding all strings inserted into any page template (a troublesome feature causing a lot of work). I'm sorry I can't assist with a PR...

**Comment by k-pd** [comment]
Forgotten screenshots:

![console](https://user-images.githubusercontent.com/26119310/42569793-0558ba48-8512-11e8-94f4-b0c0c1830271.png)
![network](https://user-images.githubusercontent.com/26119310/42569794-0577b1b4-8512-11e8-8663-7dfdf4ac1d7f.png)

**Comment by oliverklee** [comment]
HI @k-pd, thanks for the bug report. Could you please post the corresponding part of the HTML? Thanks!

**Comment by k-pd** [comment]
Excerpt from the answer to `http://localhost:8083/seminar/register/?tx_seminars_pi1[seminar]=1&tx_seminars_pi1[action]=register` sent as POST after submitting second page of registration:

```HTML
<div class="tx-seminars-pi1-hidden">
    <input type="hidden" name="tx_seminars_pi1_registration_editor[seats]" id="tx_seminars_pi1_registration_editor__seats" value="1" />
    <input type="hidden" name="tx_seminars_pi1_registration_editor[registered_themselves]" id="tx_seminars_pi1_registration_editor__registered_themselves" value="0" />
    <input type="hidden" name="tx_seminars_pi1_registration_editor[attendees_names]" id="tx_seminars_pi1_registration_editor__attendees_names" value="me me" />
    <input type="hidden" name="tx_seminars_pi1_registration_editor[structured_attendees_names]" id="tx_seminars_pi1_registration_editor__structured_attendees_names" value="[[&amp;quot;me&amp;quot;,&amp;quot;me&amp;quot;,&amp;quot;&amp;quot;,&amp;quot;&amp;quot;]]" />
    <input type="hidden" name="tx_seminars_pi1_registration_editor[known_from]" id="tx_seminars_pi1_registration_editor__known_from" value="" />
</div>
```

**Comment by djangones** [comment]
I can confirm this problem in Typo3 8.7.x where the same configuration in Typo3 7.6.x worked fine.
Apparently, all special characters within the extension form fields in Typo3 8.7.x are encoded accordingly before being output again.
For example, the ampersand in the company name of the billing address is encoded again and again, which leads to ampersand chains (&amp;amp;amp;amp;) if you navigate back and forth several times...

Translated with www.DeepL.com/Translator

**Comment by djangones** [comment]
I can confirm this problem in Typo3 8.7.x where the same configuration in Typo3 7.6.x worked fine.
Apparently, all special characters within the extension form fields in Typo3 8.7.x are encoded accordingly before being output again.
For example, the ampersand in the company name of the billing address is encoded again and again, which leads to ampersand chains (&amp;amp;amp;amp;) if you navigate back and forth several times...

**Comment by oliverklee** [comment]
I have been able to reproduce this problem both in TYPO3 7.6 as well as 8.7. This is a bug in the mkforms extension. I'll report a bug there.

**Comment by oliverklee** [comment]
This is the bug: https://github.com/DMKEBUSINESSGMBH/typo3-mkforms/issues/84

**Comment by oliverklee** [comment]
Time spend debugging and fixing so far: 55min

**Comment by oliverklee** [comment]
This should be fixed with the latest mkforms release.

**Comment by djangones** [comment]
I have checked this again and the error does not occur with the following configuration in Typo3 7.6:
seminars: 2.0.1
mkforms: 3.0.14
oelib: 2.0.1
With the same configuration in Typo3 8.7.18 the problem occurs.

Then I checked again if there are further dependencies and found out that mk_forms uses rn_base in Typo3 7.6 in version 1.6.1. After updating it to the current rn_base 1.8.2, the problem also occurs in Typo3 7.6.
Conversely, the problem no longer occurs in Typo3 8.7 when rb_base is downgraded from 1.8.2 to 1.6.1.

Obviously rb_base in combination with mkforms seems to be the problem here.
Anywas, mk_form https://github.com/DMKEBUSINESSGMBH/typo3-mkforms/commit/82dc001d9c225aef7ac8f31d8adec42ea00eea95 solves the problem.
